extensions/hadoop/src/test/java/com/hazelcast/jet/hadoop/file/PathAndGlobFileSourceTest.java

```diff
@@ -51,6 +51,7 @@ public void shouldReadFilesMatchingGlobInTheMiddle() {
 
     @Test
     public void shouldReadFilesMatchingGlobInPath() {
+        assumeThatNoWindowsOS(); // * is not allowed in filename
```

Comment:
Why shouldn't this test run on windows? I understand why `shouldReadFileWithEscapedGlob` doesn't run because of * in filename is not allowed on windows, but here the * is in the path, matching any directory.

Reply:
I don't know, maybe hadoop supports glob chars, but it fails on [this line](https://github.com/hazelcast/hazelcast-jet/pull/2746/files#diff-7e44fe2919d2b846158485b1e0f5ec4ae9a388b2201f9e51613a5d287a35b60aR98) in `Paths.get`. `Paths` doesn't support glob characters. If this is the case, we have to check that the directory is absolute in some other way. Also, if I'm not mistaken, globs are matched differently on windows and linux, and java's `FileSystem.getPathMatcher()` specifies its own syntax (maybe it's that of linux, i don't know). 

Secondly, I don't know what should the `*` mean. The test is disabled for local file system - because in local file system we read only a single directory - will hadoop read multiple directories? Also, this behavior isn't specified:
```
     * The path must point to a directory. All files in the directory are
     * processed. The directory is not processed recursively.
```
IMO we should disallow this behavior...

Reply:
Ok, it seems quite a mess. I have some bugs reported for the file connector, so will send it with those fixes.